ethereum/rpc/namespaces/eth/api.go

```diff
@@ -859,3 +888,19 @@ func (e *PublicAPI) GetProof(address common.Address, storageKeys []string, block
 		StorageProof: storageProofs,
 	}, nil
 }
+
+// getBlockNumber returns the BlockNumber from BlockNumberOrHash
+func (e *PublicAPI) getBlockNumber(blockNrOrHash rpctypes.BlockNumberOrHash) (rpctypes.BlockNumber, error) {
+	var blockNum = rpctypes.EthEarliestBlockNumber
+	if blockNrOrHash.BlockHash != nil {
+		blockHeader, err := e.backend.HeaderByHash(*blockNrOrHash.BlockHash)
+		if err != nil {
+			return blockNum, err
+		}
+		blockNum = rpctypes.NewBlockNumber(blockHeader.Number)
+	} else if blockNrOrHash.BlockNumber != nil {
```

Comment:
please use `switch` with the following cases:

`BlockHash == nil && BlockNumber == nil` -->  `error`
`BlockHash != nil `
`BlockNumber != nil`
`default : EthLatestBlockNumber`

Reply:
done